src/main/kotlin/net/atos/zac/flowable/cmmn/AanvullendeInformatieTaskListener.kt

```diff
@@ -0,0 +1,95 @@
+package net.atos.zac.flowable.cmmn
+
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.zac.configuratie.ConfiguratieService.Companion.STATUSTYPE_OMSCHRIJVING_AANVULLENDE_INFORMATIE
+import net.atos.zac.configuratie.ConfiguratieService.Companion.STATUSTYPE_OMSCHRIJVING_INTAKE
+import net.atos.zac.flowable.FlowableHelper
+import net.atos.zac.flowable.processengine.ProcessEngineLookupImpl.getCmmnEngineConfiguration
+import nl.lifely.zac.util.NoArgConstructor
+import org.flowable.cmmn.api.runtime.PlanItemInstance
+import org.flowable.engine.delegate.TaskListener
+import org.flowable.task.api.TaskInfo
+import org.flowable.task.service.delegate.BaseTaskListener.EVENTNAME_COMPLETE
+import org.flowable.task.service.delegate.BaseTaskListener.EVENTNAME_CREATE
+import org.flowable.task.service.delegate.DelegateTask
+import java.util.logging.Logger
+
+@NoArgConstructor
+class AanvullendeInformatieTaskListener : TaskListener {
+
+    companion object {
+        private val LOG = Logger.getLogger(AanvullendeInformatieTaskListener::class.java.name)
+
+        private const val STATUS_TOELICHTING = "Status gewijzigd"
+        private const val TASK_AANVULLENDE_INFORMATIE_ID = "AANVULLENDE_INFORMATIE"
+    }
+
+    override fun notify(delegateTask: DelegateTask) {
+        LOG.fine {
+            "AanvullendeInformatie task ${delegateTask.id} in state ${delegateTask.state} was ${delegateTask.eventName}"
+        }
+        (delegateTask as TaskInfo).let { taskInfo ->
+            when (delegateTask.eventName) {
+                EVENTNAME_CREATE -> createdEvent(taskInfo.subScopeId)
+                EVENTNAME_COMPLETE -> completedEvent(taskInfo.scopeId, taskInfo.subScopeId)
+            }
+        }
+    }
+
+    private fun createdEvent(subScopeId: String) {
+        updateParentZaakStatus(getPlanItemInstance(subScopeId), STATUSTYPE_OMSCHRIJVING_AANVULLENDE_INFORMATIE)
+    }
+
+    private fun completedEvent(scopeId: String, subScopeId: String) {
+        if (numberOfAdditionalInfoTasks(scopeId) == 1) {
+            updateParentZaakStatus(getPlanItemInstance(subScopeId), STATUSTYPE_OMSCHRIJVING_INTAKE)
+        }
+    }
+
+    private fun getPlanItemInstance(subScopeId: String): PlanItemInstance =
+        getCmmnEngineConfiguration().getCmmnRuntimeService()
+            .createPlanItemInstanceQuery()
+            .planItemInstanceId(subScopeId)
+            .singleResult()
+
+    private fun numberOfAdditionalInfoTasks(scopeId: String) =
+        getCmmnEngineConfiguration().getCmmnRuntimeService()
+            .createPlanItemInstanceQuery()
+            .caseInstanceId(scopeId)
+            .planItemDefinitionId(TASK_AANVULLENDE_INFORMATIE_ID)
+            .planItemInstanceState("active")
+            .list().size
+
+    private fun updateParentZaakStatus(planItemInstance: PlanItemInstance, statustypeOmschrijving: String) =
```

Comment:
What do you mean by 'parent zaak'? The zaak which belongs to the task I think? I would leave out the word 'parent' then because that is usually used for parent-child relationships between zaken when a parent zaak has a child zaak.